rules/header_paths.bzl

```diff
@@ -0,0 +1,55 @@
+load("@bazel_skylib//lib:paths.bzl", "paths")
+
+def _stringify_mapping(headers_mapping):
+    """
+    Convert a mapping of {Target: string} to a mapping of {string:string}
+
+    - Target is a target pointing to a single source file
+    - the value in `headers_mapping` is the path within Headers/PrivateHeaders to use
+      for the file.
+    """
+    ret = {}
+    for (t, dest) in headers_mapping.items():
+        files = t.files.to_list()
+        if len(files) != 1:
+            fail("{} should be a single file".format(t))
+        f = files[0]
+        if not f.is_source:
+            fail("{} should be a single source file, not a generated file".format(f))
+        ret[f.owner.name] = dest
```

Comment:
So, it *should*, in our use cases, generally be the same. But if we're doing all of this validation (1 file, that it's a source, etc), I just decided to use the owner on the File object, rather than the label on the Target object. It's Technically More Correct, but I believe they should be identical.